packages/rum/src/requestCollection.ts

```diff
@@ -45,33 +46,41 @@ interface CustomXhrContext extends XhrContext {
   requestIndex: number
 }
 
-export function trackXhr([requestStartObservable, requestCompleteObservable]: RequestObservables, tracer: Tracer) {
+export function trackXhr(
+  configuration: Configuration,
+  [requestStartObservable, requestCompleteObservable]: RequestObservables,
+  tracer: Tracer
+) {
   const xhrProxy = startXhrProxy<CustomXhrContext>()
   xhrProxy.beforeSend((context, xhr) => {
-    const tracingResult = tracer.traceXhr(context, xhr)
-    if (tracingResult) {
-      context.traceId = tracingResult.traceId
-      context.spanId = tracingResult.spanId
-    }
-    context.requestIndex = getNextRequestIndex()
+    if (isAllowedRequestUrl(configuration, context.url!)) {
```

Comment:
We could specify a proper type for `context` instead of `Partial<XhrContext>` in this case: we now that we'll always have `startTime`/`url`/`method`, a no other property